src/utils/communication_buffer.hpp

```diff
@@ -407,10 +438,10 @@ bool CommBuffer<T>::TryReceive() noexcept {
 
 template <class T>
 void CommBuffer<T>::Stale() {
-  PARTHENON_REQUIRE(*comm_type_ != BuffCommType::sender, "Should never get here.");
+  // PARTHENON_REQUIRE(*comm_type_ != BuffCommType::sender, "Should never get here.");
 
-  if (!(*state_ == BufferState::received || *state_ == BufferState::received_null))
-    PARTHENON_DEBUG_WARN("Staling buffer not in the received state.");
+  // if (!(*state_ == BufferState::received || *state_ == BufferState::received_null))
+  //   PARTHENON_DEBUG_WARN("Staling buffer not in the received state.");
```

Comment:
Why is it OK now to stale a stale'd buffer?

Reply:
In the per variable-boundary communication model, we never explicitly staled send buffers. Rather, we just checked if they were available for write (i.e. that their previous MPI communication was finished) before reusing them for the next step. For coalesced communication, these buffers still exist but they are never actually used for MPI communication. So after their data is copied into the coalesced buffer, we need to set their state to stale by hand to signify they are safe to use on the next step. There may be an argument that I need to clean up the communication buffer state logic though...

Reply:
OK that makes sense, thanks, although in that case should these commented out lines just be deleted?

Reply:
yep, now deleted.